backend/BackendMIIT/src/main/java/com/example/BackendMIIT/repository/DirectionRepository.java

```diff
@@ -1,13 +1,25 @@
 package com.example.BackendMIIT.repository;
 
 import com.example.BackendMIIT.model.domain.Direction;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface DirectionRepository extends BaseRepository<Direction> {
+
     Optional<Direction> findByName(String name);
 
     Optional<Direction> findByCode(String code);
+
+    @Query("SELECT d FROM Direction d LEFT JOIN d.passPoints p ORDER BY p.avg DESC, d.name ASC")
+    List<Direction> findAllOrderByAvgPassPointAndName();
+
+    @Query("SELECT d FROM Direction d LEFT JOIN d.passPoints p ORDER BY p.min DESC, d.name ASC")
+    List<Direction> findAllOrderByMinPassPointAndName();
+
+    @Query("SELECT d FROM Direction d ORDER BY d.name ASC")
+    List<Direction> findAllOrderByName();
```

Comment:
Use DSL instead. You can sort data like that
![image](https://github.com/user-attachments/assets/3100cb1b-2044-4c21-9364-68bc86f0fe07)

Reply:
Use findAll and delegate sorting to the service method using Sort